packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -199,25 +214,33 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
             title: VSXExtensionsCommands.INSTALL_FROM_VSIX.dialogLabel,
             openLabel: nls.localizeByDefault('Install from VSIX'),
             filters: { 'VSIX Extensions (*.vsix)': ['vsix'] },
-            canSelectMany: false
+            canSelectMany: false,
+            canSelectFiles: true
         };
         const extensionUri = await this.fileDialogService.showOpenDialog(props);
         if (extensionUri) {
             if (extensionUri.path.ext === '.vsix') {
-                const extensionName = this.labelProvider.getName(extensionUri);
-                try {
-                    await this.commandRegistry.executeCommand(VscodeCommands.INSTALL_FROM_VSIX.id, extensionUri);
-                    this.messageService.info(nls.localizeByDefault('Completed installing {0} extension from VSIX.', extensionName));
-                } catch (e) {
-                    this.messageService.error(nls.localize('theia/vsx-registry/failedInstallingVSIX', 'Failed to install {0} from VSIX.', extensionName));
-                    console.warn(e);
-                }
+                await this.installVSIX(extensionUri);
             } else {
                 this.messageService.error(nls.localize('theia/vsx-registry/invalidVSIX', 'The selected file is not a valid "*.vsix" plugin.'));
             }
         }
     }
 
+    /**
+     * Installs a local extension file.
+     */
+    protected async installVSIX(fileURI: URI): Promise<void> {
```

Comment:
I think we should also move the guarding into this method, just to avoid any obvious errors:

```
if (extensionUri.path.ext === '.vsix') {
   ...
}
} else {
   this.messageService.error(nls.localize('theia/vsx-registry/invalidVSIX', 'The selected file is not a valid "*.vsix" plugin.'));
}
```

What do you think?

Reply:
As  this method can be used only programmatically and *.vsix is just a zip file I didn't add a file extension guard here. Docu comment says 'Installs a local extension file.'  so theoretically it can be used to install from a Zip file with any extension.

Reply:
Didn't think about that but you are right, if no other code is testing the extension a simple zip might also work. Our other actions do restrict this though so if we keep it like that (at which I'm not against), please add a short comment explaining why we do not do the check here.